### ISSUE TITLE
ImportError: cannot import name 'int_classes' from 'torch._six' (Future Torch 1.9+ Compatibility Fix Required)

### ISSUE THREAD

**Issue by etetteh** [ivadomed/ivadomed#1094]
### Current behavior

* If we execute ivadomed training on Torch 1.9+ we run into this particular import error, mainly because most existing code bases are only tested with Torch 1.8.X. 
   * A potential solution has been [this](https://discuss.pytorch.org/t/cannot-import-name-int-classes-from-torch-six/126611) as a quick fix to make the code base support Torch 1.9.x. 

### Expected behavior

* Expected a successful run but instead received a `ImportError: cannot import name 'int_classes' from 'torch._six' ` error on the Mac Monterey Platform with Torch 1.10

### Steps to reproduce

* Execute `ivadomed -c config.json --train --path-data data_example_spinegeneric --path-output testRun` on a Torch 1.9+ environment with the existing master branch as of 2022-03-13

  <details>
     <summary> An example summary message </summary>
      
      ```
      Traceback (most recent call last):
        File "/Users/en_tetteh/miniforge3/envs/venv_ivadomed/bin/ivadomed", line 33, in <module>
          sys.exit(load_entry_point('ivadomed', 'console_scripts', 'ivadomed')())
        File "/Users/en_tetteh/miniforge3/envs/venv_ivadomed/bin/ivadomed", line 25, in importlib_load_entry_point
          return next(matches).load()
        File "/Users/en_tetteh/miniforge3/envs/venv_ivadomed/lib/python3.8/importlib/metadata.py", line 77, in load
          module = import_module(match.group('module'))
        File "/Users/en_tetteh/miniforge3/envs/venv_ivadomed/lib/python3.8/importlib/__init__.py", line 127, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
        File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
        File "<frozen importlib._bootstrap>", line 991, in _find_and_load
        File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 843, in exec_module
        File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
        File "/Users/en_tetteh/NEUROPOLY/ivadomed/ivadomed/main.py", line 14, in <module>
          from ivadomed import evaluation as imed_evaluation
        File "/Users/en_tetteh/NEUROPOLY/ivadomed/ivadomed/evaluation.py", line 9, in <module>
          from ivadomed import inference as imed_inference
        File "/Users/en_tetteh/NEUROPOLY/ivadomed/ivadomed/inference.py", line 15, in <module>
          from ivadomed.loader.mri3d_subvolume_segmentation_dataset import MRI3DSubVolumeSegmentationDataset
        File "/Users/en_tetteh/NEUROPOLY/ivadomed/ivadomed/loader/mri3d_subvolume_segmentation_dataset.py", line 8, in <module>
          from ivadomed import transforms as imed_transforms, postprocessing as imed_postpro
        File "/Users/en_tetteh/NEUROPOLY/ivadomed/ivadomed/transforms.py", line 19, in <module>
          from ivadomed.loader import utils as imed_loader_utils
        File "/Users/en_tetteh/NEUROPOLY/ivadomed/ivadomed/loader/utils.py", line 11, in <module>
          from torch._six import string_classes, int_classes
      ImportError: cannot import name 'int_classes' from 'torch._six' (/Users/en_tetteh/miniforge3/envs/venv_ivadomed/lib/python3.8/site-packages/torch/_six.py)
      ```
   </details>

### System description

* macOS Monterey - version 12.2.1 (21D62) with torch 1.10+
    
    <details open>
    <summary> Output of <code>pip freeze</code> </summary>
    
    ```
    absl-py==1.0.0
    astor==0.8.1
    attrs==21.4.0
    bids-neuropoly==0.2
    bids-validator==1.9.2
    cached-property @ file:///home/conda/feedstock_root/build_artifacts/cached_property_1615209429212/work
    cachetools==5.0.0
    certifi==2021.10.8
    charset-normalizer==2.0.12
    click==8.0.4
    csv-diff==1.1
    cycler==0.11.0
    Deprecated==1.2.13
    dictdiffer==0.9.0
    docopt==0.6.2
    fonttools==4.29.1
    formulaic==0.2.4
    google-auth==2.6.0
    google-auth-oauthlib==0.4.6
    grpcio @ file:///Users/runner/miniforge3/conda-bld/grpcio_1645230394268/work
    h5py @ file:///Users/runner/miniforge3/conda-bld/h5py_1637964070496/work
    humanize==4.0.0
    idna==3.3
    imageio==2.16.1
    importlib-metadata==4.11.2
    iniconfig==1.1.1
    interface-meta==1.2.4
    -e git+https://github.com/etetteh/ivadomed.git@8d1e36368a5e6ccceb1fd82f56e4b7b15ddd4bdf#egg=ivadomed
    joblib==1.1.0
    kiwisolver==1.3.2
    loguru @ file:///Users/runner/miniforge3/conda-bld/loguru_1635858978966/work
    Markdown==3.3.6
    matplotlib==3.5.1
    networkx==2.7
    nibabel==3.2.2
    num2words==0.5.10
    numpy @ file:///Users/runner/miniforge3/conda-bld/numpy_1643958864831/work
    oauthlib==3.2.0
    packaging==21.3
    pandas==1.4.1
    Pillow==9.0.1
    pluggy==1.0.0
    protobuf==3.19.4
    py==1.11.0
    pyasn1==0.4.8
    pyasn1-modules==0.2.8
    pybids==0.14.0
    pyparsing @ file:///home/conda/feedstock_root/build_artifacts/pyparsing_1635267989520/work
    pytest==7.0.1
    python-dateutil==2.8.2
    pytz==2021.3
    PyWavelets==1.2.0
    requests==2.27.1
    requests-oauthlib==1.3.1
    rsa==4.8
    scikit-image==0.19.2
    scikit-learn==1.0.2
    scipy==1.8.0
    seaborn==0.11.2
    SimpleITK==2.1.1
    six @ file:///home/conda/feedstock_root/build_artifacts/six_1620240208055/work
    SQLAlchemy==1.3.24
    tensorboard==2.8.0
    tensorboard-data-server==0.6.1
    tensorboard-plugin-wit==1.8.1
    threadpoolctl==3.1.0
    tifffile==2022.2.9
    tomli==2.0.1
    torch==1.10.2
    torchio==0.18.74
    torchvision==0.11.3
    tqdm==4.63.0
    typing_extensions==4.1.1
    urllib3==1.26.8
    Werkzeug==2.0.3
    wrapt==1.13.3
    zipp==3.7.0
    ```
    
    </details>

### Discussion

* Per PyTorch  forum discussion [here](https://discuss.pytorch.org/t/cannot-import-name-int-classes-from-torch-six/126611/2), it seems that IF the environment is Torch==1.9+ (as evident by the pip freeze here showing torch==1.10.2, and the thread linked reporting 1.9), while the codebase of ivadomed is mostly tested against Torch 1.8, it will result in this error. This is not a huge surprise (as ivadomed specifically requires 1.8). 

However, the purpose of this issue is to bring people's attention to this necessary change in the future event when we need to * upgrade to Torch 1.9+, that we need to make the necessary change recommended in the thread which is simply: 

* Other resources showing the similar problem: 
  * https://stackoverflow.com/a/69170538
  * https://blog.csdn.net/u010684651/article/details/121018127 (solution shown in English as the last few lines)

**Comment by jcohenadad** [comment]
@etetteh can you please indicate why this issue was closed? has it been resolved? if so how?

**Comment by etetteh** [comment]
The issue is due to the usage of a version of torch higher than what is required for the current ivadomed version.  The issue is not resolved, but not relevant at this time. However, it serves as a guide for the amendment of ivadomed codebase.

**Comment by dyt811** [comment]
@etetteh I have updated your issue a bit to provide a broader context and how and when it will become important. I agree the fix suggested seems rather minor and should not have major consequences so we can open a PR and patch this out to pave the way fur Torch 1.9+ adoption in the near future. 

BTW, @etetteh is there any particular reason you used torch 1.10 instead of the 1.8 recommended? Was it just for purely experimental reason to leverage the latest PyTorch 1.10 features? or was it a hardware requirement?

**Comment by etetteh** [comment]
The use of `torch==1.10` was purely for experimentation. Specifically, the `1.10` version has new `lr schedulers` that are currently being used to update `torchvision` models, which I was experimenting with.

I think you can raise the priority so we incorporate it ASAP, before "it slips through the cracks"

**Comment by dyt811** [comment]
> Also, I would like to know if there is a particular reason, why we are using torch version `1.8`. Based on my experimentation so far, everything runs fine with `1.10`

The only reason was for reproducibility and validated performance and we did [a thorough testing to ensure 1.8 is working okay](https://github.com/ivadomed/ivadomed/pull/819) at the time so we locked to that version. If we do the same for 1.10 and incorporate the latest changes, I think that is not a problem. After all, the jump from 1.5 to 1.8 was bigger than 1.9/1.10 I believe.

**Comment by dyt811** [comment]
Closed by #1104. @etetteh please ensure to mention the source issue with `resolves` attribute to auto close the issue PR was related. This is another important reason to examine the PR template. It tackles all these details. :) 

See all keywords available in the github context here: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
